unit-tests/tasks/fortran_test.py

```diff
@@ -607,46 +658,3 @@ def test_preprocssor_output(self, caplog, tmp_path):
                    FUNCTION included_when_test_macro_not_set()
                    IMPLICIT NONE
                    END FUNCTION included_when_test_macro_not_set''')
-
-
-class TestFortranCompiler(object):
-    def test_constructor(self):
-        test_unit = FortranCompiler(Path('input.f90'),
-                                    Path('workspace'),
-                                    ['flag1', 'flag2'],
-                                    [Path('prereq1.mod'),
-                                     Path('prereq2.mod')])
-        assert test_unit.input == [Path('prereq1.mod'),
-                                   Path('prereq2.mod'),
-                                   Path('input.f90')]
-        assert test_unit.output == [Path('workspace/input.o')]
-        assert test_unit.as_list == ['gfortran', '-c', '-Jworkspace',
-                                     'flag1', 'flag2', 'input.f90',
-                                     '-o', 'workspace/input.o']
-
-
-class TestFortranLinker(object):
-    def test_constructor(self):
-        test_unit = FortranLinker(Path('workspace'),
-                                  ['flag3', 'flag4'],
-                                  Path('bin/output'))
-        test_unit.add_object(Path('an.o'))
-        assert test_unit.input == [Path('an.o')]
-        assert test_unit.output == [Path('bin/output')]
-        assert test_unit.as_list == ['gfortran', '-o', 'bin/output',
-                                     'flag3', 'flag4', 'an.o']
-
-    def test_add_object(self):
-        test_unit = FortranLinker(Path('deepspace'),
-                                  [],
-                                  Path('bin/dusty'))
-        with pytest.raises(TaskException):
-            _ = test_unit.as_list
-
-        test_unit.add_object(Path('foo.o'))
-        assert test_unit.as_list == ['gfortran', '-o', 'bin/dusty',
-                                     'foo.o']
-
-        test_unit.add_object(Path('bar.o'))
-        assert test_unit.as_list == ['gfortran', '-o', 'bin/dusty',
-                                     'foo.o', 'bar.o']
```

Comment:
Are we no longer testing these things?

Reply:
I think we ought to still test the classes, but the previous tests were only really confirming that the old `Command` objects would correctly format the command.  Under the new design I can't think of how you would test them (we obviously don't want to run a compile / link to do so) - maybe there's something clever that can be done with mock/dummy objects to override `subprocess.run` so that instead of running something it returns the command it would have run and we can check it?

Reply:
Given that this stuff is now happening in the engine it may fall under the testing for the engine.